src/settings.js

```diff
@@ -0,0 +1,567 @@
+            title: "Setting one title",
+            description: "Setting one description",
+            icon: "../icons/gear-cloud-black.png",
+            value: "b"
+        }, {
+            path: "settingTwoPath",
+            type: "string",
+            values: ["b", "c", "d", "e"],
+            title: "Setting two title",
+            description: "Setting two description",
+            icon: "../icons/gear-cloud-black.png",
+            value: "c"
+        }, {
+            path: "textfieldPath",
+            type: "text",
+            title: "Text input",
+            description: "Text input description",
+            icon: "../icons/gear-cloud-white.png",
+            value: ""
+        }, {
+            path: "invertColorsPath",
+            type: "boolean",
+            title: "Invert colors",
+            description: "Invert colors description",
+            icon: "../icons/gear-cloud-black.png",
+            value: true
+        }, {
+            path: "highContrastPath",
+            type: "radio",
+            title: "High contrast",
+            description: "High contrast description",
+            icon: "../icons/gear-cloud-white.png",
+            value: false
+        }, {
+            path: "zoomPath",
+            type: "number",
+            title: "Zoom",
+            description: "Zoom description",
+            icon: "../icons/gear-cloud-black.png",
+            value: 1,
+            min: 0.5,
+            max: 4,
+            divisibleBy: 0.1
+        }, {
+            path: "spacingPath",
+            type: "numberStep",
+            title: "Spacing",
+            description: "Spacing description",
+            icon: "../icons/gear-cloud-black.png",
+            value: 1,
+            min: 0.5,
+            max: 2,
+            divisibleBy: 0.25
+        }, {
+            path: "ttsTrackingPath",
+            type: "array",
+            title: "TTS tracking mode",
+            description: "TTS tracking mode description",
+            icon: "../icons/gear-cloud-white.png",
+            values:  ["mouse", "caret", "focus"],
+            value: ["mouse", "focus"]
+        }]);
+    };
+
+    gpii.app.settings.hasPreferenceSets = function (preferenceSets) {
+        // TODO use in settingRow
+        return preferenceSets && preferenceSets.names && preferenceSets.names.length > 0;
+    };
+
+    gpii.app.settings.updateSetting = function (path, value) {
+        ipcRenderer.send("updateSetting", {
+            path: path,
+            value: value
+        });
+    };
+
+    gpii.app.settings.keyOut = function () {
+        ipcRenderer.send("keyOut");
+    };
+
+    gpii.app.settings.closeSettingsWindow = function () {
+        ipcRenderer.send("closeSettingsWindow");
+    };
+
+    fluid.registerNamespace("gpii.app.settings.settingRow");
+
+    gpii.app.settings.settingRow.getWidgetOptions = function (widgetGrade, model) {
+        switch (widgetGrade) {
+        case "gpii.app.settings.widgets.dropDown": {
```

Comment:
Stick these in grade definitions rather than a giant switch statement.

Reply:
These could go into the "exemplar" grades that we talk about below - or perhaps into a 3rd hierarchy (a bit messy)